### PR TITLE
UNR-322 : Add more profiling coverage

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Increased unreal profiling coverage within the SpatialGDK plugin.
Based on stat captures provided by existing games.

#### Primary reviewers
@danielimprobable @improbable-danny

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -1540,7 +1551,6 @@ void USpatialReceiver::ProcessRPCEventField(Worker_EntityId EntityId, const Work
 			{
 				ObjectRef.Entity = Schema_GetEntityId(EventData, SpatialConstants::UNREAL_PACKED_RPC_PAYLOAD_ENTITY_ID);
 
-
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Is this intentional?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 2]
```diff
@@ -70,7 +74,6 @@ void USpatialReceiver::Init(USpatialNetDriver* InNetDriver, FTimerManager* InTim
 
 void USpatialReceiver::OnCriticalSection(bool InCriticalSection)
 {
-	SCOPE_CYCLE_COUNTER(STAT_ReceiverCritSection);
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: We no longer profiling for EnterCriticalSection, will we still want this?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp [hunk 3]
```diff
@@ -166,112 +169,118 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
...
+							{
+								RootObjectReferencesMap.Remove(SwappedCmd.Offset);
+							}
+							bOutReferencesChanged = true;
 						}
-						bOutReferencesChanged = true;
+					}
+					else
+					{
+						ApplyArray(ComponentObject, FieldId, RootObjectReferencesMap, ArrayProperty, Data, SwappedCmd.Offset, ShadowOffset, Cmd.ParentIndex, bOutReferencesChanged);
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp]: Do we also want to track this and ApplyProperty below?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Yeah - too many blank lines

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp (hunk 3 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp]: Sure

**Comment by m-samiec** [comment]
> Overall looks good to me, just a couple of questions:
> 
> Do we want to do any additional tracking in RPC container? Do we want to keep the changes that I did for MW there too? Finally, do we want to guard some of these stats behind flags so that we can enable/disable them to avoid them messing with the execution time tracking unnecessarily?

I don't think we need to profile the RPC container, as it's just a utility class that calls into the sender/receiver where necessary. The functions it calls in those classes SendRPC/ApplyRPC have had profiling added. And all these stats are behind a few compiler flags already (mostly ENABLE_STATNAMEDEVENTS), so I think that should be sufficient.

**Pull request merged**